### PR TITLE
Add documentation for the site-config.json keys

### LINKED ISSUE #470: Write a doc for site-config.json
The `site-config.json` in a client project file contains the configuration settings required for a gisida site. The `site-config.json` has grown to support keys that are used for certain configurations such as having the search box display which does not by default, loading the splash page as an iframe. A doc for this file explaining mandatory keys, optional keys and what each attribute is meant for should be created

### PR DESCRIPTION
**Pull request by kelvin-muchiri**
Closes #470 

Documentation for the `site-config.json` keys

Add documentation for the keys:

- [x] APP
- [x] APP.oauthclientID
- [x] APP.mapConfig
- [x] APP.mapConfig.container
- [x] APP.mapConfig.style
- [x] APP.mapConfig.center
- [x] APP.mapConfig.zoom
- [x] APP.accessToken
- [x] APP.appIcon
- [x] APP.appName
- [x] APP.appColor
- [x] APP.appNameDesc
- [x] APP.appLoginIcon
- [x] APP.searchBar
- [x] APP.mapStateToUrl
- [x]  APP.hasNavBar
- [ ] APP.supersetOnlyLogin
- [x] APP.mapConfig.mapBounds
- [x] APP.mapConfig.preserveDrawingBuffer
- [x] APP.apiAccessToken
- [x] APP.includeNavControls
- [x] APP.removeMapScale
- [x] APP.supersetBase
- [x] APP.authConfig
- [x] APP.authConfigApiMap
- [x] APP.showSinglePeriods
- [x] APP.layerSupersetLinks
- [x] APP.NULL_LAYER_TEXT
- [x] SUPERSET_CONFIGS
- [x] SUPERSET_CONFIGS.iframes
- [x] SUPERSET_CONFIGS.splashPage
- [x] STYLES
- [x] LAYERS
- [x] APP.hasNavBar

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+        }
+      }
+      ```
+  - `preserveDrawingBuffer` : whether to preserve the buffers until manually cleared or overwritten.
+
+- `accessToken` : Your Mapbox API access token.
+- `hyperLink` : Add links and description to menu items. 
+- `authConfigApiMap` : Supplemental auth api string, can either be data or metadata.
+- `apiAccessToken` : Ona data/external api access token.
+- `includeNavControls` : Boolean value determines if map controls will be included.
```
Comment (kelvin-muchiri) [inline README.md]: > `includeNavControls` : Boolean value determines if map controls will be included.

 Let's mention whether it's optional and if so, what's the default value if the key is not provided

File: README.md [hunk 2]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+- `removeMapScale` : Show or hide map scale.
```
Comment (kelvin-muchiri) [inline README.md]: > `removeMapScale` : Show or hide map scale.

Let's mention it's a boolean if it is, whether it's optional and if so, what's the default value if the key is not provided

File: README.md [hunk 3]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+          "lng": 62.89563867185916,
+          "lat": 11.188804478371793
+        }
+      }
+      ```
+  - `preserveDrawingBuffer` : whether to preserve the buffers until manually cleared or overwritten.
+
+- `accessToken` : Your Mapbox API access token.
+- `hyperLink` : Add links and description to menu items. 
+- `authConfigApiMap` : Supplemental auth api string, can either be data or metadata.
```
Comment (kelvin-muchiri) [inline README.md]: > authConfigApiMap`: Supplemental auth API string, can either be data or metadata.

Let's add what it is used for and if it's optional

**Comment by kelvin-muchiri** [comment]
@kahummer I was thinking we should against each key if **(Required**) or **(Optional)** and those that are optional to appear after the required. e.g

`accessToken` **(Required)**: Your Mapbox API access token.

File: README.md [hunk 4]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+- `apiAccessToken` : Ona data/external api access token.
+- `includeNavControls` : Boolean value determines if map controls will be included.
+- `removeMapScale` : Show or hide map scale.
+- `showSinglePeriods` : Show timeseries element for single periods.
```
Comment (kelvin-muchiri) [inline README.md]: > `showSinglePeriods` : Show timeseries element for single periods.

Let's mention it's a boolean if it is, whether it's optional and if so, what's the default value if the key is not provided

File: README.md [hunk 5]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+- `NULL_LAYER_TEXT` : Custom Text to be shown when user has no access to a layer.
```
Comment (kelvin-muchiri) [inline README.md]: Let's whether it's optional and if so, what's the default value if the key is not provided

File: README.md [hunk 6]
```diff
@@ -262,24 +220,225 @@ We can also pull data from different forms and join the various form data. Here
...
+        "_ne": {
+          "lng": 62.89563867185916,
+          "lat": 11.188804478371793
+        }
+      }
+      ```
+  - `preserveDrawingBuffer` : whether to preserve the buffers until manually cleared or overwritten.
+
+- `accessToken` : Your Mapbox API access token.
+- `hyperLink` : Add links and description to menu items.
```
Comment (kelvin-muchiri) [inline README.md]: > `hyperLink` : Add links and description to menu items. 

Let's mention each of the keys for this object and what they are used for such as `mapConfig` object above

File: README.md [hunk 7]
```diff
@@ -262,24 +220,226 @@ We can also pull data from different forms and join the various form data. Here
...
+- `authConfigApiMap` **(Optional)** : Supplemental auth api string, can either be data or metadata. Defaults to metadata 
+    e.g https://api.ona.io/api/v1/data/467840.csv or https://api.ona.io/api/v1/metadata/467840.csv
+- `apiAccessToken` **(Optional)** : Ona data/external api access token.
+- `includeNavControls` **(Optional)** : Boolean value determines if map controls will be included. Defaults to true
+- `removeMapScale` **(Optional)** : Removes map scale. Defaults to false
+- `showSinglePeriods` **(Optional)** : Show timeseries for single periods. Defaults to false
+- `NULL_LAYER_TEXT` **(Optional)** : Custom Text to be shown when user has no access to a layer. Defaults to ``` You don't have permision to view this category```
+- `appIcon` **(Optional)** : The relative path to image that will be displayed as the logo on the top of the title bar.
+- `appName` **(Optional)** : The name for the app that will be displayed on the title bar.
+- `appColor` **(Required)** : The app primary color that will be displyed most frequently across the screens such
```
Comment (kelvin-muchiri) [inline README.md]: > `appColor` **(Required)** : The app primary color that will be displyed most frequently across the screens such

Let's have all required before the optional
**displyed**  Let's correct the typo

File: README.md [hunk 8]
```diff
@@ -262,24 +220,226 @@ We can also pull data from different forms and join the various form data. Here
...
+        "_ne": {
+          "lng": 62.89563867185916,
+          "lat": 11.188804478371793
+        }
+      }
+      ```
+  - `preserveDrawingBuffer` : whether to preserve the buffers until manually cleared or overwritten.
+
+- `accessToken` **(Required)** :  Your Mapbox basemap style API access token.
+- `hyperLink` **(Optional)** : Add links and description to menu items.
```
Comment (kelvin-muchiri) [inline README.md]: Let's add what each property in the `hyperLink` object does, just like we did for `mapConfig` above. Let's also add an example and a link to the full documentation in **gisida-react** such as we have done for the documentation on properties:

`APP.layerSupersetLinks`
`SUPERSET_CONFIGS.iframes`
`SUPERSET_CONFIGS.splashPage`

below
Screenshot from 2020-09-25 12-39-05

**Review by kahummer — APPROVED** [review]

**Review by ukanga — APPROVED** [review]

**Review by samwata — APPROVED** [review]

**Pull request merged**